docs/internals/controls-package.md

```diff
@@ -9,20 +9,280 @@ description: "The Control Package implements feedback control on an endpoint."
+
+**Returns:**
+
+- [(error)](https://pkg.go.dev/builtin#error): An error, if one occurred.
+
+```go
+// add necessary attributes to the PIDLoop struct
+pidLoop := &PIDLoop{}
+
+err := pidLoop.StartControlLoop()
+```
+
+### CreateConstantBlock
+
+Creates a new control block of type `constant`.
+
+**Parameters:**
+
+- `ctx` [(Context)](https://pkg.go.dev/context): A Context carries a deadline, a cancellation signal, and other values across API boundaries.
+- `name` [(string)](https://pkg.go.dev/builtin#string): The desired name of the constant block.
+- `constVal` [(float64)](https://pkg.go.dev/builtin#float64): The value of the new set point.
+
+**Returns:**
+
+[(BlockConfig)](/internals/controls-package/#blockconfig): The config for the newly created block.
+
+```go
+constBlock := control.CreatConstantBlock(context.Background(), "set_point", 10.0)
+```
+
+### UpdateConstantBlock
+
+Creates a new control block of type `constant`, and then updates the control loop to use this new block.
+
+**Parameters:**
+
+- `ctx` [(Context)](https://pkg.go.dev/context): A context carries a deadline, a cancellation signal, and other values across API boundaries.
+- `name` [(string)](https://pkg.go.dev/builtin#string): The desired name of the constant block.
+- `constVal` [(float64)](https://pkg.go.dev/builtin#float64): The value of the new set point.
+- `loop` (`*Loop`): The control loop that needs to be updated with the newly created constant block.
+
+**Returns:**
+
+- [(error)](https://pkg.go.dev/builtin#error): An error, if one occurred.
+
+```go
+err := control.UpdateConstantBlock(context.Background(), "set_point", 10.0, loop)
+```
+
+### CreateTrapzBlock
+
+Creates a new control block of type `trapezoidalVelocityProfile`.
+
+**Parameters:**
+
+- `ctx` [(Context)](https://pkg.go.dev/context): A Context carries a deadline, a cancellation signal, and other values across API boundaries.
+- `name` [(string)](https://pkg.go.dev/builtin#string): The desired name of the trapezoidal block.
+- `maxVel` [(float64)](https://pkg.go.dev/builtin#float64): The max velocity for the controlled component to move at.
+- `dependsOn` [([]string)](https://pkg.go.dev/builtin#string): A slice strings containing the names of the other control blocks that the new trapezoidal block depends on. Usually the set point and the end point.
+
+**Returns:**
+
+[(BlockConfig)](/internals/controls-package/#blockconfig): The config for the newly created block.
+
+```go
+trapzBlock := control.CreateTrapzBlock(context.Background(), "set_point", 10.0, []string{"set_point", "endpoint"})
+```
+
+### UpdateTrapzBlock
+
+Creates a new control block of type `trapezoidalVelocityProfile`, and then updates the control loop to use this new block.
+
+**Parameters:**
+
+- `ctx` [(Context)](https://pkg.go.dev/context): A context carries a deadline, a cancellation signal, and other values across API boundaries.
+- `name` [(string)](https://pkg.go.dev/builtin#string): The desired name of the trapezoidal block.
```

Comment:
it's the name for the block that's being created by the function. so it could be anything. to replace an old block the name must be the same but if creating a new one completely it can be named anything

Reply:
Ok but whatever I specify here will be used and not changed by other code? Then we want to remove "desired" because it can imply that you're only specifying a preference

Reply:
yeah we don't want it to be changed it should be the same every time... that make sense